src/language/printer.js

```diff
@@ -31,7 +31,8 @@ const printDocASTReducer = {
     const defs = wrap('(', join(node.variableDefinitions, ', '), ')');
     const directives = join(node.directives, ' ');
     const selectionSet = node.selectionSet;
-    return !name ? selectionSet :
+    return !name ? !directives && !defs && (!op || op === `query`) ?
```

Comment:
@leebyron thanks for the good catch - especially that `query` with directives/variable defs would ignore the artifacts and only show the selectionSet, treating it as a short-handed query. Added explicit checks for that exact condition, as well as tests for it.